src/app/pages/apps/components/custom-app-form/custom-app-form.component.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  ChangeDetectionStrategy, ChangeDetectorRef, Component,
+  OnInit,
+} from '@angular/core';
+import { Validators } from '@angular/forms';
+import { FormBuilder } from '@ngneat/reactive-forms';
+import { UntilDestroy, untilDestroyed } from '@ngneat/until-destroy';
+import { TranslateService } from '@ngx-translate/core';
+import { map } from 'rxjs';
+import { CodeEditorLanguage } from 'app/enums/code-editor-language.enum';
+import { Role } from 'app/enums/role.enum';
+import { AppCreate } from 'app/interfaces/app.interface';
+import { DialogService } from 'app/modules/dialog/dialog.service';
+import { IxSlideInRef } from 'app/modules/forms/ix-forms/components/ix-slide-in/ix-slide-in-ref';
+import { forbiddenAsyncValues } from 'app/modules/forms/ix-forms/validators/forbidden-values-validation/forbidden-values-validation';
+import { ApplicationsService } from 'app/pages/apps/services/applications.service';
+import { ErrorHandlerService } from 'app/services/error-handler.service';
+import { WebSocketService } from 'app/services/ws.service';
+
+@UntilDestroy()
+@Component({
+  selector: 'ix-custom-app-form',
+  templateUrl: './custom-app-form.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class CustomAppFormComponent implements OnInit {
+  protected requiredRoles = [Role.AppsWrite];
+  protected readonly CodeEditorLanguage = CodeEditorLanguage;
+  form = this.fb.group({
+    release_name: ['', Validators.required],
+    custom_compose_config_string: ['', Validators.required],
+  });
+  isLoading = false;
+  protected tooltip = this.translate.instant('Add custom app config in Yaml format.');
+  forbiddenAppNames$ = this.appService.getAllApps().pipe(map((apps) => apps.map((app) => app.name)));
+  constructor(
+    private fb: FormBuilder,
+    private translate: TranslateService,
+    private cdr: ChangeDetectorRef,
+    private ws: WebSocketService,
+    private errorHandler: ErrorHandlerService,
+    private dialogService: DialogService,
+    private appService: ApplicationsService,
+    private dialogRef: IxSlideInRef<CustomAppFormComponent>,
+  ) {}
+
+  ngOnInit(): void {
+    this.addForbiddenAppNamesValidator();
```

Comment:
I've noticed that the form could not pass validation when there's no installed apps. I'd double check the validation logic.